File: go/problems/day21_test.go
package problems

import "testing"

var day21 Solvable[int] = Day21{}

func TestDay21_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day21_1")
	result := day21.Solve(input)

	var expected int = 29
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay21_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day21_2")
	result := day21.Solve(input)

	var expected int = 3776
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day21.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import solve from "./day21.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 21", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day21_1");
    const result = solve(input);
    assert.equal(result, 29);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day21_2");
    const result = solve(input);
    assert.equal(result, 3776);
  });
});
